[Fixit Q4 2022] Add official splunk repository + bump cdap and hadoop…

feefs-google opened the pull request:
There were [permissions changes](https://spring.io/blog/2020/10/29/notice-of-permissions-changes-to-repo-spring-io-fall-and-winter-2020) for `repo.spring.io` that caused fetching artifacts to fail. Added splunk's jfrog repository as suggested in the [Splunk Java SDK repository](https://github.com/splunk/splunk-sdk-java).

<img width="579" alt="1" src="https://user-images.githubusercontent.com/114443254/207175869-a0fa4204-d984-4796-8832-f0163fc32c7a.png">
log4j is now only in the `test` scope

<img width="430" alt="2" src="https://user-images.githubusercontent.com/114443254/207176017-0961cede-5db5-4c6d-aafa-2ed3d581fc61.png">
Build and unit tests passing

dli357 commented on pom.xml:
```diff
@@ -61,6 +61,10 @@
       <id>spring</id>
```
Let's go ahead and remove the spring repo as it's no longer used.

dli357 commented on pom.xml:
```diff
@@ -208,6 +212,16 @@
       <artifactId>spark-streaming_2.11</artifactId>
       <version>${spark2.version}</version>
       <scope>provided</scope>
+      <exclusions>
```
It seems like jets3t was cut from spark 2.4: https://issues.apache.org/jira/browse/SPARK-23654, so I think adding an exclusion is ok. Is bumping the splunk dependency an option?

dli357 commented on pom.xml:
```diff
@@ -29,9 +29,9 @@
     <!-- version properties -->
-    <cdap.version>6.1.2</cdap.version>
+    <cdap.version>6.8.0</cdap.version>
     <hydrator.version>2.3.0-SNAPSHOT</hydrator.version>
```
Please bump this to 2.10.0, thanks!

feefs-google commented:
Addressed PR comments, project still builds and passes all unit tests:
<img width="428" alt="Screen Shot 2022-12-12 at 3 32 39 PM" src="https://user-images.githubusercontent.com/114443254/207182206-6983f95d-be44-41e3-b016-066c4be49d2c.png">

feefs-google commented:
Bumped spark version to 2.4.8, meaning we don't need to add the exclusion anymore. Project still builds and passes unit tests:
<img width="437" alt="Screen Shot 2022-12-12 at 3 53 57 PM" src="https://user-images.githubusercontent.com/114443254/207184781-d58ede18-461a-4117-bf99-b8cb4c59233a.png">

dli357 commented:
LGTM 👍
